vortex/core/helpers.py

```diff
@@ -27,3 +27,37 @@ def input_size(job):
 def weighted_random_sampling(destinations):
     rankings = [(d.params.get('weight', 1) if d.params else 1) for d in destinations]
     return random.choices(destinations, weights=rankings, k=len(destinations))
+
+
+def get_keys_from_dict(dl, keys_list):
+    # This function builds a list using the keys from nest dictionaries
+    # (copied from galaxyproject/galaxy lib/galaxy/jobs/dynamic_tool_destination.py)
+    if isinstance(dl, dict):
+        keys_list.extend(dl.keys())
+        for x in dl.values():
+            get_keys_from_dict(x, keys_list)
+    elif isinstance(dl, list):
+        for x in dl:
+            get_keys_from_dict(x, keys_list)
+
+
+def job_arguments_match(job, app, arguments):
```

Comment:
Only so it's a little shorter
```suggestion
def job_args_match(job, app, args):
```